openedx/core/djangoapps/user_authn/views/password_reset.py

```diff
@@ -215,3 +239,227 @@ def request_password_change(email, is_secure):
     else:
         # No user with the provided email address exists.
         raise errors.UserNotFound
+
+
+@csrf_exempt
+@require_POST
+def password_reset(request):
+    """
+    Attempts to send a password reset e-mail.
+    """
+    # Add some rate limiting here by re-using the RateLimitMixin as a helper class
+    limiter = BadRequestRateLimiter()
+    if limiter.is_rate_limit_exceeded(request):
+        AUDIT_LOG.warning("Rate limit exceeded in password_reset")
+        return HttpResponseForbidden()
+
+    form = PasswordResetFormNoActive(request.POST)
+    if form.is_valid():
+        form.save(use_https=request.is_secure(),
+                  from_email=configuration_helpers.get_value('email_from_address', settings.DEFAULT_FROM_EMAIL),
+                  request=request)
+        # When password change is complete, a "edx.user.settings.changed" event will be emitted.
+        # But because changing the password is multi-step, we also emit an event here so that we can
+        # track where the request was initiated.
+        tracker.emit(
+            SETTING_CHANGE_INITIATED,
+            {
+                "setting": "password",
+                "old": None,
+                "new": None,
+                "user_id": request.user.id,
+            }
+        )
+        destroy_oauth_tokens(request.user)
+    else:
+        # bad user? tick the rate limiter counter
+        AUDIT_LOG.info("Bad password_reset user passed in.")
+        limiter.tick_request_counter(request)
+
+    return JsonResponse({
+        'success': True,
+        'value': render_to_string('registration/password_reset_done.html', {}),
+    })
+
+
+def uidb36_to_uidb64(uidb36):
```

Comment:
Nit: Throughout the file, consider adding `_` prefix for methods that are private.  Not sure if this one is, but it seems like it should be. :)